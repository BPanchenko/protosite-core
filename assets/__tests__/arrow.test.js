import path from 'node:path'

const { BASE_DIR, BASE_URL } = global.server
const testURL = new URL(
	path.relative(BASE_DIR, __filename).replace('.js', '.html'),
	BASE_URL,
)

describe('ArrowComponent', () => {
	let page, screenshot

	beforeAll(async () => {
		page = await global.browser.newPage()
		await page.setViewport({ width: 768, height: 1024 })
		await page.goto(testURL)

		screenshot = Buffer.from(
			await page.screenshot({
				fullPage: true,
				encoding: 'base64',
			}),
			'base64',
		)
	}, 15000)

	it('Page Markup', () => expect(page.content()).resolves.toMatchSnapshot())

	describe('Visual Regression', () => {
		it('horizontal', () => {
			expect(screenshot).toMatchImageSnapshot({
				diffDirection: 'horizontal',
			})
		})
		it('vertical', () => {
			expect(screenshot).toMatchImageSnapshot({
				diffDirection: 'vertical',
			})
		})
	})
})
