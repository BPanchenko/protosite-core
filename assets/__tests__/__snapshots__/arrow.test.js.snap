// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowComponent Page Markup 1`] = `
"<!DOCTYPE html><html><head>
		<title>[TEST] Arrow Component ‚Åï Protosite Core</title>
		<link href="http://assets.protosite.rocks/uikit/document.css" rel="stylesheet" type="text/css">
		<link href="http://assets.protosite.rocks/uikit/objects.css" rel="stylesheet" type="text/css">

		<link href="http://assets.protosite.rocks/uikit/component/arrow.css" rel="stylesheet" type="text/css">
		<script src="../arrow-component.mjs" type="module"></script>
	</head>
	<body class="t-document">
		<main class="o-container" data-testid="container">
			<section class="c-panel s-clean">
				<c-arrow></c-arrow>
			</section>
		</main>
	

</body></html>"
`;
